### DISCUSSION TITLE
Linter ignoring typescript errors

### DISCUSSION THREAD

**Discussion by Pacheco95** [vercel/next.js#34802] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
/bin/sh: 1: pnpm: not found

    Operating System:
      Platform: linux
      Arch: x64
      Version: #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022
    Binaries:
      Node: 16.12.0
      npm: 8.1.1
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.1.0
      react: 17.0.2
      react-dom: 17.0.2

warn  - Latest canary version not detected, detected: "12.1.0", newest: "12.1.1-canary.2".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
```

### Describe the Bug

Running `npm run lint` (`next lint`) does not check for typescript errors

### Expected Behavior

Running `npm run lint` (`next lint`) to report typescript errors in console

### To Reproduce

Create a bare new project with typescript template and force some TS errors then run `npm run lint`. The errors are not shown.

**Comment by timneutkens** [comment]
Converted to a feature request because it's functioning as documented: https://nextjs.org/docs/api-reference/cli#lint

> `next lint` runs ESLint for all files in the pages, components, and lib directories.
